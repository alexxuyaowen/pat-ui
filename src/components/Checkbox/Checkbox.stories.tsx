import React from 'react';
import Checkbox from './Checkbox';

export default {
  title: 'Checkbox',
  component: Checkbox,
};

export const BasicCheckbox = () => (
  <div className="checkboxes">
    <Checkbox label="Default Checked" defaultChecked />
    <Checkbox label="Default" />
    {/* <Checkbox label="Indeterminate" /> */}
    <Checkbox label="Disabled" disabled />
    <Checkbox label="Disabled Checked" disabled checked />
  </div>
);

export const DiffSizeCheckbox = () => (
  <div className="checkboxes">
    <Checkbox label="Small" size="small" defaultChecked />
    <Checkbox label="Default" defaultChecked />
    <Checkbox label="Large" size="large" defaultChecked />
  </div>
);

export const DiffColorCheckbox = () => (
  <div className="checkboxes">
    <Checkbox label="Primary" defaultChecked />
    <Checkbox label="Secondary" color="secondary" defaultChecked />
    <Checkbox label="Success" color="success" defaultChecked />
    <Checkbox label="Default" color="default" defaultChecked />
  </div>
);

export const DiffIconCheckbox = () => (
  <div className="checkboxes">
    <Checkbox icon="heart" />
    <Checkbox icon="check" />
    <Checkbox icon="star" />
    <Checkbox icon="moon" />
  </div>
);
