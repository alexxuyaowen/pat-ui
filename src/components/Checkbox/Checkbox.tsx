import React, { FC, useState } from 'react';
import { classNames } from '../../utils/classNames';
import Icon from '../Icon';

export type CheckboxSize = 'small' | 'large';
export type CheckboxColor = 'secondary' | 'success' | 'default';

export interface ICheckboxProps {
  defaultChecked?: boolean;
  disabled?: boolean;
  checked?: boolean;
  label?: string;
  size?: CheckboxSize;
  className?: string;
  color?: CheckboxColor;
  icon?: string;
}

const Checkbox: FC<ICheckboxProps> = (props) => {
  const {
    disabled,
    checked,
    defaultChecked,
    label,
    size,
    className,
    color,
    icon,
  } = props;

  const [check, setCheck] = useState(defaultChecked || checked);

  let styleClasses = classNames('checkbox', {
    [`checkbox-${color}`]: !!color,
    [`checkbox-${size}`]: !!size,
  });
  if (className) {
    styleClasses += ' ' + className;
  }

  const onChangeHandler = () => {
    setCheck((prev) => !prev);
  };

  return (
    <div className={styleClasses}>
      {icon ? (
        <React.Fragment>
          <input
            className="checkbox-icon"
            type="checkbox"
            disabled={disabled}
            checked={check}
            onChange={onChangeHandler}
          />
          <Icon name={icon} />
        </React.Fragment>
      ) : (
        <React.Fragment>
          <input
            type="checkbox"
            disabled={disabled}
            checked={check}
            onChange={onChangeHandler}
          />
          <label>{label}</label>
        </React.Fragment>
      )}
    </div>
  );
};

Checkbox.defaultProps = {
  disabled: false,
  checked: false,
  defaultChecked: false,
};

export default Checkbox;
