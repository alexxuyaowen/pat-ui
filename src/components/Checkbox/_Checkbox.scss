.checkboxes {
  display: flex;
  justify-content: center;
  gap: 5vw;
}

input[type='checkbox'] {
  cursor: pointer;
}

input[type='checkbox']:disabled {
  cursor: not-allowed;
}
