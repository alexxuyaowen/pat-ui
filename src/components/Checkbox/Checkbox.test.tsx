import React from 'react';
import { render } from '@testing-library/react';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Checkbox />);
    expect(asFragment()).toMatchSnapshot();
  });
});
