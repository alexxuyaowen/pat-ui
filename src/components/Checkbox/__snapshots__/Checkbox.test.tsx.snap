// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="checkbox"
  >
    <input
      type="checkbox"
    />
    <label />
  </div>
</DocumentFragment>
`;
